### PR TITLE
refactor: Use Traefik for proxying Vault web traffic instead of Caddy

### PR DESCRIPTION
**Pull request by blarghmatey**
The Caddy plugin for managing Let's Encrypt DNS challenges with Route53 is broken, despite some fixes being applied. Traefik is a more widely used and adaptable proxy service that also includes built-in support for Let's Encrypt, sowe are migrating to use it in place of Caddy.

## Description
This replaces the usage of Caddy on the Vault servers with Traefik

## Motivation and Context
The current builds of Vault are broken due to the lack of certificate handling in Caddy.

## How Has This Been Tested?
Applied the Vault deploy to a local Vagrant box to ensure that it runs cleanly.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (improves on existing behavior)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project. (Did you install and run the pre-commit hooks?)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/bilder/images/vault/deploy.py [hunk 1]
```diff
@@ -51,18 +57,22 @@
 # Set up configuration objects
 set_env_secrets(Path("consul/consul.env"))
 # Install Caddy
```
Comment (Ardiea) [inline src/bilder/images/vault/deploy.py:59]:
```suggestion
# Install Traefik
```

File: src/bilder/images/vault/deploy.py [hunk 2]
```diff
@@ -51,18 +57,22 @@
...
-caddy_config = CaddyConfig(
-    plugins=[
-        CaddyPlugin(
-            repository="github.com/caddy-dns/route53",
-            version=VERSIONS["caddy_route53"],
+traefik_config = TraefikConfig(
+    static_configuration=traefik_static.TraefikStaticConfig.parse_obj(
+        yaml.safe_load(
+            FILES_DIRECTORY.joinpath("traefik", "static_config.yaml").read_text()
```
Comment (Ardiea) [inline src/bilder/images/vault/deploy.py:63]: Did you omit these two files from the commit by accident or am I misunderstanding how the component works?

File: src/bilder/images/vault/files/traefik/static_config.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    http:
+      redirections:                           # HTTPS redirection (80 to 443)
+        entryPoint:
+          to: "vault"                         # The target element
+          scheme: "https"
+  vault:
+    address: ":443"
+providers:
+  file:
+    filename: /etc/traefik/vault.yaml
```
Comment (Ardiea) [inline src/bilder/images/vault/files/traefik/static_config.yaml:14]: Is this a file that we need to provide as well? Or did this show up with traefik? I presume it looks something like [this](https://doc.traefik.io/traefik-enterprise/providers/vault-kv/#provider-configuration).

Comment (blarghmatey) [reply src/bilder/images/vault/files/traefik/static_config.yaml:14]:
```
vagrant@bullseye:~$ sudo ls /etc/traefik/*.yaml
/etc/traefik/traefik.yaml  /etc/traefik/vault.yaml
```

Comment (blarghmatey) [reply src/bilder/images/vault/files/traefik/static_config.yaml:14]:
```
vagrant@bullseye:~$ cat /etc/traefik/vault.yaml 
{
  "http": {
    "routers": {
      "vault-router": {
        "rule": "Host({{ env \"DOMAIN\"}})",
        "service": "vault-service",
        "tls": {
          "cert_resolver": "letsencrypt_resolver"
        }
      },
      "vault-healthcheck": {
        "rule": "PathPrefix(\"/v1/sys/health\")",
        "service": "vault-service",
        "tls": {
          "cert_resolver": "letsencrypt_resolver"
        }
      }
    },
    "services": {
      "vault-service": {
        "load_balancer": {
          "servers": [
            {
              "url": "https://localhost:8200"
            }
          ],
          "servers_transport": "vault-transport"
        }
      }
    }
  }
}
```

Comment (Ardiea) [reply src/bilder/images/vault/files/traefik/static_config.yaml:14]: 👍 I follow now.

**Review by Ardiea — APPROVED** [review]
👍

**Pull request merged**